src/store/flow/block.ts

```diff
@@ -74,9 +97,35 @@ export const mutations: MutationTree<IFlowsState> = {
     findBlockExitWith(exitId, block)
       .destination_block = destinationBlockId
   },
+  block_updateVendorMetadataByPath(state, {blockId, path, value}: { blockId: string, path: string, value: object | string }) {
+    // ensure reactivity
+    const metadata = findBlockOnActiveFlowWith(blockId, state as unknown as IContext).vendor_metadata
+    const pathAsList = toPath(path)
+
+    // todo: I have no idea what TS is complaining about here --- "No matching method overload"
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    // eslint-disable-next-line lodash/collection-method-value
+    reduce(pathAsList, (accumulator, pathKey, i): void => {
+      const nextValue = i === pathAsList.length - 1 ? value : get(accumulator, pathKey, {})
+      Vue.set(accumulator, pathKey, nextValue)
+      return nextValue
+    }, metadata!)
+  },
 }
 
 export const actions: ActionTree<IFlowsState, IRootState> = {
+  block_popExitsByLabel({ dispatch, commit, state }, { blockId, exitLabel, shouldUseCache = false }: {blockId: string; exitLabel: string; shouldUseCache: boolean}) {
+    const block = findBlockOnActiveFlowWith(blockId, state as unknown as IContext)
+    const newBlockExits = findBlockExitsRef(block, shouldUseCache).filter((item: IBlockExit) => item.label !== exitLabel)
```

Comment:
We no longer have label. I don't know exactly how we're meant to link exits to choices now?